src/wellKnownDidConfiguration/wellKnownDidConfiguration.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  Did,
+  CType,
+  Credential,
+  Claim,
+  SignCallback,
+  DidUri,
+  Utils,
+  ICredentialPresentation,
+} from '@kiltprotocol/sdk-js'
+import {
+  CredentialSubject,
+  DomainLinkageCredential,
+  VerifiableDomainLinkagePresentation,
+} from '../types/types'
+import * as validUrl from 'valid-url'
+
+export const DEFAULT_VERIFIABLECREDENTIAL_TYPE = 'VerifiableCredential'
+export const KILT_VERIFIABLECREDENTIAL_TYPE = 'KiltCredential2020'
+export const KILT_SELF_SIGNED_PROOF_TYPE = 'KILTSelfSigned2020'
+
+export const ctypeDomainLinkage = CType.fromSchema({
+  $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+  title: 'Domain Linkage Credential',
+  properties: {
+    id: {
+      type: 'string',
+    },
+    origin: {
+      type: 'string',
+    },
+  },
+  type: 'object',
+})
+
+export async function createCredential(
+  signCallback: SignCallback,
+  origin: string,
+  didUri: DidUri
+): Promise<ICredentialPresentation> {
+  const fullDid = await Did.resolve(didUri)
+
+  if (!fullDid?.document) {
+    throw new Error('No Did found: Please create a Full DID')
+  }
+
+  const { document } = fullDid
+
+  if (!validUrl.isUri(origin)) {
+    throw new Error('The origin is not a valid url')
+  }
+
+  const domainClaimContents = {
+    id: document.uri,
+    origin,
+  }
+
+  const claim = Claim.fromCTypeAndClaimContents(
+    ctypeDomainLinkage,
+    domainClaimContents,
+    document.uri
+  )
+
+  const credential = Credential.fromClaim(claim)
+
+  const assertionKey = document.assertionMethod?.[0]
+
+  if (!assertionKey) {
+    throw new Error(
+      'Full DID doesnt have assertion key: Please add assertion key'
+    )
+  }
+
+  return Credential.createPresentation({
+    credential,
+    signCallback,
+  })
+}
+
+export async function getDomainLinkagePresentation(
+  credential: ICredentialPresentation,
+  expirationDate: string = new Date(
+    Date.now() + 1000 * 60 * 60 * 24 * 365 * 5
+  ).toISOString()
+): Promise<VerifiableDomainLinkagePresentation> {
+  const claimContents = credential.claim.contents
+  if (!claimContents.id && !claimContents.origin) {
+    throw new Error('Claim contents do not content an id or origin')
+  }
+
+  Did.validateUri(claimContents.id)
+
+  const didUri = claimContents.id as DidUri
+
+  let origin: string
+  if (typeof claimContents.origin !== 'string') {
+    throw new Error('claim contents id is not a string')
+  } else if (!validUrl.isUri(claimContents.origin)) {
+    throw new Error('The claim contents origin is not a valid url')
+  } else {
+    origin = claimContents.origin
+  }
+
+  const credentialSubject: CredentialSubject = {
+    id: didUri,
+    origin,
+    rootHash: credential.rootHash,
```

Comment:
is that in the spec? I'm surprised to see this property here even though it is not in the CType

Reply:
Unfortunately, we don't have that in the current context. I spoke with Antonio and he will make the custom context at some point.

Reply:
I have removed the `rootHash` and added the rootHash as the `Id` in the document.